### PR TITLE
Added `Ladder_Variational_Autoencoders_jax.ipynb`

### PR DESCRIPTION
**Pull request by susnato**
### Description
Implemented `Ladder_Variational_Autoencoders` in Jax.

### Colab link

https://colab.research.google.com/drive/11mGSvbrNkfQl6WNODZ44vU40wEP1XhGX?usp=sharing

### Issue

https://github.com/probml/pyprobml/issues/681

### Checklist:

[✔]Performed a self-review of the code.
[✔]Tested on Google Colab.

### CODE REVIEW

**Comment by susnato** [comment]
Can you please review my code? @murphyk @mjsML @gerdm

**Comment by codeboy5** [comment]
Hey @susnato, did you implement `ConvTranspose2d` or just use `nn.ConvTranspose`. As per the docs there is notable difference between both.

According to the docs :- 

`torch.nn.ConvTranspose2d` and `nn.ConvTranspose` are not compatible. `nn.ConvTranspose` is a wrapper around `jax.lax.conv_transpose` which computes a fractionally strided convolution, while `torch.nn.ConvTranspose2d` computes a gradient based transposed convolution. 

It is currently an open issue in [Flax Issue](https://github.com/google/flax/issues/1872), so i think we should be aware and when that pr is merged, this code should also be changed.

Let me know what you think about this.

**Comment by susnato** [comment]
Hey @codeboy5, I used the of nn.ConvTranspose.
Since Flax does not have support for the `gradient based transposed convolution` I preferred to go with that, btw I will keep an eye on that PR you mentioned and chance the code in future.

**Comment by murphyk** [comment]
It would be useful to add some text comments cells to this notebook to explain what is going on, and perhaps add some diagrams. Also, maybe a toy example in 2d to illustrate bottom up and top down inference would be more enlightening than just generating pixels?

**Comment by murphyk** [comment]
It seems that https://github.com/probml/pyprobml/pull/746 also implements ladder VAE, so I am closing current issue.
